src/common/oclengine.cpp

```diff
@@ -52,12 +55,34 @@ void OCLEngine::SetDeviceContextPtrVector(std::vector<DeviceContextPtr> vec, Dev
 
 void OCLEngine::SetDefaultDeviceContext(DeviceContextPtr dcp) { default_device_context = dcp; }
 
-OCLEngine::OCLEngine() { InitOCL(); }
+OCLEngine::OCLEngine()
+{
+    OCLInitResult res = InitOCL(false);
+    all_device_contexts = res.all_device_contexts;
+    default_device_context = res.default_device_context;
+}
 OCLEngine::OCLEngine(OCLEngine const&) {}
 OCLEngine& OCLEngine::operator=(OCLEngine const& rhs) { return *this; }
 
-void OCLEngine::InitOCL()
+OCLInitResult OCLEngine::InitOCL(bool saveBinaries)
 {
+#if defined(_WIN32) && !defined(__CYGWIN__)
+    char* homeDrive = getenv("HOMEDRIVE");
+    char* homePath = getenv("HOMEPATH");
+    int newSize = strlen(homeDrive) + strlen(homePath) + 1;
+    char* home = new char[newSize];
+    strcpy(home, homeDrive);
+    strcat(home, homePath);
+    std::string homeStr(home);
```

Comment:
Use a ostringformat or a snprintf.  Check to see if the getenv calls succeed.  I believe macOS programs (opened from the GUI) don’t get HOME set - check this.

Reply:
This has been simplified. Mac does set HOME, but it requires sudo to write to it, but I think not to read from it.